Friendly-plans/app/src/main/java/pg/autyzm/friendly_plans/validation/Validation.java

```diff
@@ -0,0 +1,68 @@
+package pg.autyzm.friendly_plans.validation;
+
+import android.widget.EditText;
+import java.util.regex.Pattern;
+
+/***
+ * Original of validation class is here:
+ * https://tausiq.wordpress.com/2013/01/19/android-input-field-validation/
+ */
+public class Validation {
+
+    // Regular Expression
+    private static final String NAME_REGEX = "^[a-zA-Z0-9_. -]*$";
+    private static final String NUM_ONLY_REGEX = "^[0-9]*$";
+
+    // Error Messages
+    private static final String WRONG_NAME_MSG = "@string/only_letters_msg";
+    private static final String DUPLICATED_NAME_MSG = "@string/name_exist_msg";
+    private static final String ONLY_NUMBERS_MSG = "@string/only_numbers_msg";
+    private static final String NOT_EMPTY_MSG = "@string/not_empty_msg";
```

Comment:
It's great that you moved it as Gosia asked to string.xml but it can't be used like this. It leads to displaying messages like '@string/only_letters_msg'. We should implement it this way:
http://stackoverflow.com/questions/7493287/android-how-do-i-get-string-from-resources-using-its-name
But to gain the local resource Context is needed. So I suggest changing it from static class into a dagger's singleton and inject wherever it is needed. We should init it with app context in app class - same as DaoSession